### PR TITLE
Fixes issue #23 with dynamic python 2 & 3 support.

### PR DESCRIPTION
**Pull request by foxwoods369**
Makes the changes discussed in #23 to support vim compiled with dynamic python 2 & 3 support.

### CODE REVIEW

File: autoload/pyenv/utils.vim [hunk 1]
```diff
@@ -16,7 +16,7 @@ function! pyenv#utils#import(name) abort " {{{
 endfunction " }}}
 function! pyenv#utils#system(...) abort " {{{
   let P = pyenv#utils#import('Process')
-  let stdout = call(P.system, a:000, P)
+  silent let stdout = call(P.system, a:000, P)
```
Comment (foxwoods369) [inline autoload/pyenv/utils.vim]: Needed to prevent additional spam showing up in the status line from the system call.

**Comment by lambdalisue** [comment]
lgtm!

**Comment by oblitum** [comment]
This merge turns the plugin incompatible with NeoVim. NeoVim does not recognize `'pythonx'`.

**Comment by foxwoods369** [comment]
:( sorry @oblitum. I'm not familiar enough with pythonx to know whether it's something neovim will take or has removed....it seemed from this comment that it was going to be merged in: https://github.com/neovim/neovim/issues/2971.
That being said, for now, perhaps changing
`return has('pythonx')`
to 
`return has('pythonx') || has('python2') || has('python3')`
will work?

**Comment by oblitum** [comment]
@foxwoods369 I just reverted that line locally.

**Comment by foxwoods369** [comment]
Alright - on vanilla vim the issue will remain if that line's reverted, but I think the check against all 3 should work.

**Comment by lambdalisue** [comment]
Do you need this plugin in Neovim?
I don't need this plugin anymore because I'm using Neovim.
See 
https://github.com/lambdalisue/vim-pyenv/issues/11#issuecomment-383595196

**Comment by oblitum** [comment]
@lambdalisue good to know that, I didn't. The fact is that I use a configuration that works uniformly for both vim and neovim.

**Comment by oblitum** [comment]
@foxwoods369 what should be used in return `has('pythonx') || has('python') || has('python3')`, there's no `'python2'` feature. The change for `'pythonx'` is because it's more future proof? I don't know what's its practical advantage for pull-request. Just asking.

**Comment by foxwoods369** [comment]
Yes, my bad on the python2. In vim, if you compile with dynamic python & python3 support, calling has('python') will force it to use python2 (and vice versa for python3). In my install without these changes, any python 3 venv would fail to be loaded due to version mismatch. Using has('pythonx') avoids this.

**Comment by oblitum** [comment]
Ah, interesting, good to know. FYI, you face this problem on specific operating systems because of the way Vim is built, packaged, and links to libraries in them, even for `+python/dyn` + `python3/dyn`. ArchLinux for example is known to not have such issue, hence why I can use the plugin without issues in it. I know the issue shows up in macOS and Debian based distros.

If `'pythonx'` can evade that, then nice.

**Pull request merged**